generic/tests/vlan.py

```diff
@@ -91,11 +90,21 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
                               test.log.info)
         return session.cmd_status(rem_vlan_cmd)
 
+    def find_free_port(session):
+        """
+        Returns a free port on the guest
+        """
+        check_cmd = "netstat -nultp |awk '(NR>2){print $4}' | awk -F':' '{print $NF}'"
+        used_port_names = session.cmd_output(check_cmd).strip().splitlines()
+        port_range = list(range(1, 65535))
+        free_ports = list(set(used_port_names).difference(set(port_range)))
+        return free_ports[0]
+
     def nc_transfer(test, src, dst):
         """
         Transfer file by netcat
         """
-        nc_port = utils_misc.find_free_port(1025, 5334, vm_ip[dst])
+        nc_port = find_free_port(session)
```

Comment:
Where is the free port you want to find? In src? Or dst? `session` is not unique.

Reply:
It should be the session of src, I have made changes to the code, please help to review it again.